pretyping/classops.ml

```diff
@@ -329,21 +336,17 @@ let add_coercion_in_graph (ic,source,target) =
   let ambig_paths =
     (ref [] : ((cl_index * cl_index) * inheritance_path) list ref) in
   let try_add_new_path (i,j as ij) p =
-    try
-      if Bijint.Index.equal i j then begin
-	if different_class_params i then begin
-	  let _ = lookup_path_between_class ij in
-          ambig_paths := (ij,p)::!ambig_paths
-	end
-      end else begin
-        let _ = lookup_path_between_class ij in
-        ambig_paths := (ij,p)::!ambig_paths
-      end;
+    if not (Bijint.Index.equal i j) || different_class_params i then
+      try
+        let q = lookup_path_between_class ij in
+        if not (path_convertible p q) then
+          ambig_paths := (ij,p)::!ambig_paths;
+        false
+      with Not_found -> begin
```

Comment:
Do 
~~~ocaml
match lookup_path_between_class ij with 
| q -> if not (path_convertible p q) ...
| exception Not_found -> (add_new_path uj p; true)
~~~
that way you won't catch arbitrary exceptions from the path comparator.

Reply:
I didn't know this syntax. It's nice!